src/torrent.cpp

```diff
@@ -8768,6 +8782,7 @@ namespace {
 	{
 		TORRENT_ASSERT(is_single_thread());
 		if (m_sequential_download == sd) return;
+		if (!sd && m_picker) m_picker->set_sequential_range(piece_index_t(0), piece_index_t(-1));
```

Comment:
I think you should be explicit about passing in `m_torrent_file->end_piece()` here (assuming we make it be an open range)